tests/test_integration.py

```diff
@@ -125,6 +125,42 @@ def test_integration(self):
                 index_file
             )
 
+    def test_integration_with_package(self):
```

Comment:
This is testing a lot of similar things to what exists already. Could we remove this redundancy and check that the full names are correct instead? Perhaps including a module level function as well?

It may make sense to put the existing pyexample test into a test class with a setup and teardown instead of building two sets of docs?